test-e2e/run-test.sh

```diff
@@ -81,21 +95,31 @@ echo
 # TODO: allow for running all platforms, or only certain ones
 
 # only spin for local devs running on machine to show tests aren't frozen
-if [[ ! -v CI_ENV ]]; then
+if [ -z "$CI_ENV" ]; then
   spin &
   SPIN_PID=$!
-  trap "kill -9 $SPIN_PID" `seq 0 15`
+  trap 'kill -9 $SPIN_PID' $(seq 0 15)
 fi
 
-go test -v "github.com/hashicorp/waypoint/test-e2e"
+# Run Docker tests
+go test -v "github.com/hashicorp/waypoint/test-e2e" -run "$E2E_PLATFORM"
```

Comment:
Does the env var `E2E_PLATFORM` need to be set before running this script? If so, might be helpful to document that 😄

Reply:
We could update the script to check if it's set and throw and error if it's empty too saying what's expected

Reply:
This is in the bash file
```
if [ "${E2E_PLATFORM}" != "Docker" ] && [ "${E2E_PLATFORM}" != "Kubernetes" ] && [ "${E2E_PLATFORM}" != "Ecs" ] && [ "${E2E_PLATFORM}" != "Nomad" ]; then
   echo "Environment variable 'E2E_PLATFORM' must be one of: 'Docker', 'Kubernetes', 'Ecs', 'Nomad'"
   exit 1
 fi
```

What other documentation would you like to see? A top-level comment with instructions?

Reply:
That seems pretty legit to me personally! 👍🏻